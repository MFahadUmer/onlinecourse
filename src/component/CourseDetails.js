import React from 'react';
import UserNavbar from './userNavbar';

const CourseDetails = ({
  // eslint-disable-next-line react/prop-types
  title,
  // eslint-disable-next-line no-unused-vars,react/prop-types
  uniqueKey,
  // eslint-disable-next-line no-unused-vars,react/prop-types
  userid,
  // eslint-disable-next-line react/prop-types
  details,
  // eslint-disable-next-line react/prop-types
  requirements,
  // eslint-disable-next-line react/prop-types
  difficulty,
  // eslint-disable-next-line react/prop-types
  image,
  // eslint-disable-next-line react/prop-types
  price,
  // eslint-disable-next-line react/prop-types
  check,
  // eslint-disable-next-line react/prop-types
  addFavouriteHandler,
}) => (
  <>
    <UserNavbar />
    <div>
      <div className="singleCourseDetailsContainer">
        <img className="singleCourseDetailsImage" src={image} alt={image} />
        <div className="singleCourseDetailsImageDetails">
          <div className="singleCourseDetailsAuthor">
            <div className="singleCourseDetailsAuthorImage">
              <img className="authorImage" src="https://www.ocf.berkeley.edu/~dblab/blog/wp-content/uploads/2012/01/icon-profile.png" alt={image} />
            </div>
            <div>
              <p className="authorname">
                Author Name
              </p>
              <p>
                <span className="fas fa-star starColor" />
                <span className="fas fa-star starColor" />
                <span className="fas fa-star starColor" />
                <span className="fas fa-star starColor" />
                <span className="fas fa-star" />
              </p>
            </div>
          </div>
          <div className="singleCoursePrice">
            <p>{`${price} $`}</p>
            <p>Per Month</p>
          </div>
        </div>
      </div>
      <div className="singleCourseDetails">
        <h3 className="aboutThisCourse">{title}</h3>
        <div>
          <p className="courseDetailsParagraph">{details}</p>
          <p className="courseDetailsParagraph">
            {`Difficulty => ${difficulty}`}
          </p>
          <p className="courseDetailsParagraph">
            {`Requirements => ${requirements}`}
          </p>
        </div>
      </div>
      <div className="addToFavourites">
        <button
          className="addToFavouritesLink"
          type="submit"
          onClick={() => addFavouriteHandler({
            user_id: userid,
            course_id: uniqueKey,
          })}
        >
          {check}
        </button>
      </div>
    </div>
  </>
);

export default CourseDetails;
