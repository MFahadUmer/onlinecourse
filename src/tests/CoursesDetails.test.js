import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import CourseDetails from '../component/CourseDetails';

describe('Test Suit', () => {
  const courseId = 1;
  const userId = 1;
  const title = 'React and Redux';
  const details = 'A Very Nice Course';
  const price = '100';

  it('should found the title on screen', () => {
    render(
      <CourseDetails
        key={courseId}
        uniqueKey={courseId}
        userid={userId}
        title={title}
        details={details}
        price={price}
        check="Remove From Favourites"
      />,
    );
    expect(screen.getByText('React and Redux')).toBeTruthy();
  });

  it('should found the details on screen', () => {
    render(
      <CourseDetails
        key={courseId}
        uniqueKey={courseId}
        userid={userId}
        title={title}
        details={details}
        price={price}
        check="Remove From Favourites"
      />,
    );
    expect(screen.getByText('A Very Nice Course')).toBeTruthy();
  });

  it('should match the course details component snapshot', () => {
    const tree = renderer.create(
      <CourseDetails
        key={courseId}
        uniqueKey={courseId}
        userid={userId}
        title={title}
        details={details}
        price={price}
        check="Remove From Favourites"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
