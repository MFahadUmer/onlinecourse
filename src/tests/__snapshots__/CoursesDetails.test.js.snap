// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Suit should match the course details component snapshot 1`] = `
<div>
  <div
    className="singleCourseDetailsContainer"
  >
    <img
      alt=""
      className="singleCourseDetailsImage"
      src=""
    />
    <div
      className="singleCourseDetailsImageDetails"
    >
      <div
        className="singleCourseDetailsAuthor"
      >
        <div
          className="singleCourseDetailsAuthorImage"
        >
          <img
            alt=""
            className="authorImage"
            src="https://www.ocf.berkeley.edu/~dblab/blog/wp-content/uploads/2012/01/icon-profile.png"
          />
        </div>
        <div>
          <p
            className="authorname"
          >
            Author Name
          </p>
          <p>
            <span
              className="fas fa-star starColor"
            />
            <span
              className="fas fa-star starColor"
            />
            <span
              className="fas fa-star starColor"
            />
            <span
              className="fas fa-star starColor"
            />
            <span
              className="fas fa-star"
            />
          </p>
        </div>
      </div>
      <div
        className="singleCoursePrice"
      >
        <p>
          100 $
        </p>
        <p>
          Per Month
        </p>
      </div>
    </div>
  </div>
  <div
    className="singleCourseDetails"
  >
    <h3
      className="aboutThisCourse"
    >
      React and Redux
    </h3>
    <div>
      <p
        className="courseDetailsParagraph"
      >
        A Very Nice Course
      </p>
      <p
        className="courseDetailsParagraph"
      >
        Difficulty =&gt; 
      </p>
      <p
        className="courseDetailsParagraph"
      >
        Requirements =&gt; 
      </p>
    </div>
  </div>
  <div
    className="addToFavourites"
  >
    <button
      className="addToFavouritesLink"
      name="Remove From Favourites"
      onClick={[Function]}
      type="submit"
    >
      Remove From Favourites
    </button>
  </div>
</div>
`;
